eZ/Publish/Core/Repository/ContentService.php

```diff
@@ -1917,15 +1918,22 @@ public function removeTranslation(ContentInfo $contentInfo, $languageCode)
             }
             // check if the specified translation is not the only one
             if (count($versionInfo->languageCodes) < 2) {
-                throw new BadStateException(
-                    '$languageCode',
-                    'Specified translation is the only one Content Object Version has'
-                );
+                $singleLangVersions[] = $versionInfo->versionNo;
+                continue;
             }
             // add version to the list
             $versions[] = $versionInfo->versionNo;
         }
 
+        if (!empty($singleLangVersions)) {
+            $verList = trim(implode(', ', $singleLangVersions), ', ');
```

Comment:
I'm not sure that this `trim` is necessary here?

Reply:
No it's not, for many years I operated under the assumption that this will give me trailing delimiter, so I've been doing this automatically without checking. I've probably mistaken it with another function behavior (maybe not even PHP :P)

With an embarassment fixed also in b3dbd14 ;)